oemof/solph/blocks.py

```diff
@@ -824,25 +824,25 @@ def _create(self, group=None):
 
         m = self.parent_block()
 
-        I = {n: [i for i in n.inputs][0] for n in group}
-        O = {n: [o for o in n.outputs.keys()] for n in group}
-
-        self.relation = Constraint(group, noruleinit=True)
-
-        def _input_output_relation(block):
-            for t in m.TIMESTEPS:
-                for n in group:
-                    for o in O[n]:
-                        try:
-                            lhs = m.flow[I[n], n, t] * \
-                                  n.conversion_factors[o][t]
-                            rhs = m.flow[n, o, t]
-                        except:
-                            raise ValueError("Error in constraint creation",
-                                             "source: {0}, target: {1}".format(
-                                                 n.label, o.label))
-                        block.relation.add((n, o, t), (lhs == rhs))
-        self.relation_build = BuildAction(rule=_input_output_relation)
+        # I = {n: [i for i in n.inputs][0] for n in group}
```

Comment:
@ckaldemeyer You explain, why you loop for `n.inputs` here but over `n.outputs.keys()` below? Also, why are you accessing the first input flow here, but all output flows below?

Reply:
> @ckaldemeyer could you explain why you loop over n.inputs here but over n.outputs.keys() below?

This is the standard `LinearTransformer` and I haven't written the code. Looping over `n.inputs` should result in the same as looping over `n.inputs.keys()`. I found it just more explicit to write the keys..

> Also, why are you accessing the first input flow here, but all output flows below?

Because  a `LinearTransformer` always has one input and can have multiple outflows.